core/rawdb/accessors_chain.go

```diff
@@ -686,9 +689,13 @@ func WriteBody(db kv.RwTx, hash common.Hash, number uint64, body *types.Body) er
 		TxAmount: uint32(len(body.Transactions)) + 2,
 		Uncles:   body.Uncles,
 	}
+	if len(body.Withdrawals) > 0 {
```

Comment:
It should be `if body.Withdrawals != nil {` instead in order to preserve the difference between nil and empty withdrawals.

Reply:
Good point!